src/Dovin/Builder.hs

```diff
@@ -155,12 +162,32 @@ withLocation :: Location -> GameMonad () -> GameMonad ()
 withLocation loc m = do
   p <- view envActor
 
-  local (set (envTemplate . cardLocation) (p, loc)) m
+  local (
+        set (envTemplate . cardLocation) (p, loc)
+      . set (envTemplate . cardController) p
+      . set (envTemplate . cardZone) loc
+    ) m
+
+-- | Place the created card into a specific zone.
+withZone :: Zone -> GameMonad () -> GameMonad ()
+withZone n =
+  local (
+      set (envTemplate . cardZone) n
+    . set (envTemplate . cardLocation . _2) n
+    )
 
 -- | Set the owner for the created card. If not specified, defaults to the
 -- owner of the card location.
 withOwner :: Player -> GameMonad () -> GameMonad ()
-withOwner owner = local (set envOwner (Just owner))
+withOwner = local . set envOwner . Just
```

Comment:
maybe remove this now we have proper `withController` ... no use cases for it I don't think?